--- git status ---
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/__pycache__/humanoid.cpython-310.pyc
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/humanoid.py
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/__pycache__/reach_env_cfg.cpython-310.pyc
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__init__.py
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/__init__.cpython-310.pyc
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/joint_pos_env_cfg.cpython-310.pyc
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/__pycache__/rsl_rl_ppo_cfg.cpython-310.pyc
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/rsl_rl_ppo_cfg.py
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/joint_pos_env_cfg.py
	modified:   HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/reach_env_cfg.py
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/events.out.tfevents.1748548275.hy-LOQ-15IRX9.18521.0
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.onnx
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_0.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_100.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_150.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_200.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_250.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_300.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_350.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_400.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_450.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_50.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_500.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_550.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_600.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_650.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_700.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_750.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_800.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_850.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_900.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_950.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_999.pt
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.pkl
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.yaml
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.pkl
	deleted:    HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.yaml
	deleted:    ModelAssets/.thumbs/256x256/hand.usd.png
	modified:   ModelAssets/configuration/hand_base.usd
	modified:   ModelAssets/configuration/hand_physics.usd
	modified:   ModelAssets/configuration/hand_sensor.usd
	modified:   ModelAssets/hand.usd
	modified:   hand_test.py
	deleted:    test_old.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	HumanoidRL/logs/rsl_rl/reach_humanoid_hand/
	HumanoidRL/outputs/2025-07-02/
	ModelAssets/.thumbs/256x256/Hand.usd.png
	joint.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/__pycache__/humanoid.cpython-310.pyc b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/__pycache__/humanoid.cpython-310.pyc
index b031c7c..565075f 100644
Binary files a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/__pycache__/humanoid.cpython-310.pyc and b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/__pycache__/humanoid.cpython-310.pyc differ
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/humanoid.py b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/humanoid.py
index fcf68bd..2b19112 100644
--- a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/humanoid.py
+++ b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/modelCfg/humanoid.py
@@ -6,7 +6,7 @@ from isaaclab.assets.articulation import ArticulationCfg
 
 HAND_CFG = ArticulationCfg(
     spawn=sim_utils.UsdFileCfg(
-        usd_path="/home/hy/Downloads/Humanoid/ModelAssets/hand.usd",
+        usd_path="/home/hy/Downloads/Humanoid_Wato/ModelAssets/hand.usd",
         rigid_props=sim_utils.RigidBodyPropertiesCfg(
             disable_gravity=False,
             max_depenetration_velocity=5.0,
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/__pycache__/reach_env_cfg.cpython-310.pyc b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/__pycache__/reach_env_cfg.cpython-310.pyc
index 2d621f1..6e160b1 100644
Binary files a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/__pycache__/reach_env_cfg.cpython-310.pyc and b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/__pycache__/reach_env_cfg.cpython-310.pyc differ
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__init__.py b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__init__.py
index 7478796..37d9b69 100644
--- a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__init__.py
+++ b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__init__.py
@@ -2,13 +2,13 @@ import gymnasium as gym
 from . import agents
 
 gym.register(
-    id="Isaac-Reach-UR10-v0",
+    id="Isaac-Reach-Humanoid-Hand-v0",
     entry_point="isaaclab.envs:ManagerBasedRLEnv",
     disable_env_checker=True,
     kwargs={
-        "env_cfg_entry_point": f"{__name__}.joint_pos_env_cfg:UR10ReachEnvCfg",
+        "env_cfg_entry_point": f"{__name__}.joint_pos_env_cfg:HumanoidHandReachEnvCfg",
         "rl_games_cfg_entry_point": f"{agents.__name__}:rl_games_ppo_cfg.yaml",
-        "rsl_rl_cfg_entry_point": f"{agents.__name__}.rsl_rl_ppo_cfg:UR10ReachPPORunnerCfg",
+        "rsl_rl_cfg_entry_point": f"{agents.__name__}.rsl_rl_ppo_cfg:HumanoidHandReachPPORunnerCfg",
         "skrl_cfg_entry_point": f"{agents.__name__}:skrl_ppo_cfg.yaml",
     },
 )
@@ -18,9 +18,9 @@ gym.register(
     entry_point="isaaclab.envs:ManagerBasedRLEnv",
     disable_env_checker=True,
     kwargs={
-        "env_cfg_entry_point": f"{__name__}.joint_pos_env_cfg:UR10ReachEnvCfg_PLAY",
+        "env_cfg_entry_point": f"{__name__}.joint_pos_env_cfg:HumanoidHandReachEnvCfg_PLAY",
         "rl_games_cfg_entry_point": f"{agents.__name__}:rl_games_ppo_cfg.yaml",
-        "rsl_rl_cfg_entry_point": f"{agents.__name__}.rsl_rl_ppo_cfg:UR10ReachPPORunnerCfg",
+        "rsl_rl_cfg_entry_point": f"{agents.__name__}.rsl_rl_ppo_cfg:HumanoidHandReachPPORunnerCfg",
         "skrl_cfg_entry_point": f"{agents.__name__}:skrl_ppo_cfg.yaml",
     },
 )
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/__init__.cpython-310.pyc b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/__init__.cpython-310.pyc
index 3360e9a..7e02679 100644
Binary files a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/__init__.cpython-310.pyc and b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/__init__.cpython-310.pyc differ
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/joint_pos_env_cfg.cpython-310.pyc b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/joint_pos_env_cfg.cpython-310.pyc
index 0ee3dcd..c95067a 100644
Binary files a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/joint_pos_env_cfg.cpython-310.pyc and b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/__pycache__/joint_pos_env_cfg.cpython-310.pyc differ
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/__pycache__/rsl_rl_ppo_cfg.cpython-310.pyc b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/__pycache__/rsl_rl_ppo_cfg.cpython-310.pyc
index 9f39c93..5681edd 100644
Binary files a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/__pycache__/rsl_rl_ppo_cfg.cpython-310.pyc and b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/__pycache__/rsl_rl_ppo_cfg.cpython-310.pyc differ
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/rsl_rl_ppo_cfg.py b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/rsl_rl_ppo_cfg.py
index e913a5d..ec8a196 100644
--- a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/rsl_rl_ppo_cfg.py
+++ b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/agents/rsl_rl_ppo_cfg.py
@@ -3,11 +3,11 @@ from isaaclab_rl.rsl_rl import RslRlOnPolicyRunnerCfg, RslRlPpoActorCriticCfg, R
 
 
 @configclass
-class UR10ReachPPORunnerCfg(RslRlOnPolicyRunnerCfg):
+class HumanoidHandReachPPORunnerCfg(RslRlOnPolicyRunnerCfg):
     num_steps_per_env = 24
     max_iterations = 1000
     save_interval = 50
-    experiment_name = "reach_ur10"
+    experiment_name = "reach_humanoid_hand"
     run_name = ""
     resume = False
     empirical_normalization = False
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/joint_pos_env_cfg.py b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/joint_pos_env_cfg.py
index 90496dc..5cc271e 100644
--- a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/joint_pos_env_cfg.py
+++ b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/config/HumanoidRLEnv/joint_pos_env_cfg.py
@@ -3,34 +3,85 @@ from isaaclab.utils import configclass
 import HumanoidRLPackage.HumanoidRLSetup.tasks.manipulation.mdp as mdp
 from HumanoidRLPackage.HumanoidRLSetup.tasks.manipulation.reach_env_cfg import ReachEnvCfg
 from HumanoidRLPackage.HumanoidRLSetup.modelCfg.universal_robots import UR10_CFG
+from HumanoidRLPackage.HumanoidRLSetup.modelCfg.humanoid import HAND_CFG
 
 
 @configclass
-class UR10ReachEnvCfg(ReachEnvCfg):
+class HumanoidHandReachEnvCfg(ReachEnvCfg):
     def __post_init__(self):
         # post init of parent
         super().__post_init__()
 
-        # switch robot to ur10
-        self.scene.robot = UR10_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
+        # self.scene.robot = UR10_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
+        self.scene.robot = HAND_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
+        
+        marker_scale = 0.02
+
+        self.commands.ee_pose.goal_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+        self.commands.ee_pose.current_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+
+        self.commands.ee_pose_2.goal_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+        self.commands.ee_pose_2.current_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+
+        self.commands.ee_pose_3.goal_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+        self.commands.ee_pose_3.current_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+
+        self.commands.ee_pose_4.goal_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+        self.commands.ee_pose_4.current_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+
+        self.commands.ee_pose_5.goal_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+        self.commands.ee_pose_5.current_pose_visualizer_cfg.markers["frame"].scale = (marker_scale, marker_scale, marker_scale)
+
         # override events
         self.events.reset_robot_joints.params["position_range"] = (0.75, 1.25)
         # override rewards
-        self.rewards.end_effector_position_tracking.params["asset_cfg"].body_names = ["ee_link"]
-        self.rewards.end_effector_position_tracking_fine_grained.params["asset_cfg"].body_names = ["ee_link"]
-        self.rewards.end_effector_orientation_tracking.params["asset_cfg"].body_names = ["ee_link"]
+        self.rewards.end_effector_position_tracking.params["asset_cfg"].body_names = ["TIP_B_1"]
+        self.rewards.end_effector_position_tracking_fine_grained.params["asset_cfg"].body_names = ["TIP_B_1"]
+        # self.rewards.end_effector_orientation_tracking.params["asset_cfg"].body_names = ["TIP_B_1"]
+
+        self.rewards.end_effector_2_position_tracking.params["asset_cfg"].body_names = ["TIP_B_2"]
+        self.rewards.end_effector_2_position_tracking_fine_grained.params["asset_cfg"].body_names = ["TIP_B_2"]
+        # self.rewards.end_effector_2_orientation_tracking.params["asset_cfg"].body_names = ["TIP_B_2"]
+
+        self.rewards.end_effector_3_position_tracking.params["asset_cfg"].body_names = ["TIP_B_3"]
+        self.rewards.end_effector_3_position_tracking_fine_grained.params["asset_cfg"].body_names = ["TIP_B_3"]
+        # self.rewards.end_effector_3_orientation_tracking.params["asset_cfg"].body_names = ["TIP_B_3"]
+
+        self.rewards.end_effector_4_position_tracking.params["asset_cfg"].body_names = ["TIP_B_4"]
+        self.rewards.end_effector_4_position_tracking_fine_grained.params["asset_cfg"].body_names = ["TIP_B_4"]
+        # self.rewards.end_effector_4_orientation_tracking.params["asset_cfg"].body_names = ["TIP_B_4"]
+
+        self.rewards.end_effector_5_position_tracking.params["asset_cfg"].body_names = ["TIP_B_5"]
+        self.rewards.end_effector_5_position_tracking_fine_grained.params["asset_cfg"].body_names = ["TIP_B_5"]
+        # self.rewards.end_effector_5_orientation_tracking.params["asset_cfg"].body_names = ["TIP_B_5"]
+
         # override actions
         self.actions.arm_action = mdp.JointPositionActionCfg(
             asset_name="robot", joint_names=[".*"], scale=0.5, use_default_offset=True
         )
         # override command generator body
         # end-effector is along x-direction
-        self.commands.ee_pose.body_name = "ee_link"
-        self.commands.ee_pose.ranges.pitch = (math.pi / 2, math.pi / 2)
+        self.commands.ee_pose.body_name = "TIP_B_1"
+        # self.commands.ee_pose.ranges.pitch = (math.pi / 2, math.pi / 2) # this rotate the pose
+
+        self.commands.ee_pose_2.body_name = "TIP_B_2"
+        # self.commands.ee_pose_2.ranges.pitch = (math.pi / 2, math.pi / 2)
+
+        self.commands.ee_pose_3.body_name = "TIP_B_3"
+        # self.commands.ee_pose_3.ranges.pitch = (math.pi / 2, math.pi / 2)
+
+        self.commands.ee_pose_4.body_name = "TIP_B_4"
+        # self.commands.ee_pose_4.ranges.pitch = (math.pi / 2, math.pi / 2)
+
+        self.commands.ee_pose_5.body_name = "TIP_B_5"
+        # self.commands.ee_pose_5.ranges.pitch = (math.pi / 2, math.pi / 2)
+
+        # self.commands.ee_pose.ranges.yaw = (-math.pi / 2, -math.pi / 2)
+        # self.commands.ee_pose_2.ranges.yaw = (-math.pi / 2, -math.pi / 2)
 
 
 @configclass
-class UR10ReachEnvCfg_PLAY(UR10ReachEnvCfg):
+class HumanoidHandReachEnvCfg_PLAY(HumanoidHandReachEnvCfg):
     def __post_init__(self):
         # post init of parent
         super().__post_init__()
@@ -38,4 +89,8 @@ class UR10ReachEnvCfg_PLAY(UR10ReachEnvCfg):
         self.scene.num_envs = 50
         self.scene.env_spacing = 2.5
         # disable randomization for play
-        self.observations.policy.enable_corruption = False
\ No newline at end of file
+        self.observations.policy.enable_corruption = False
+
+# (env_isaaclab) hy@hy-LOQ-15IRX9:~/Downloads/Humanoid_Wato/HumanoidRL$ PYTHONPATH=$(pwd) /home/hy/IsaacLab/isaaclab.sh -p HumanoidRLPackage/rsl_rl_scripts/train.py --task=Isaac-Reach-Humanoid-Hand-v0 --headless
+
+# (env_isaaclab) hy@hy-LOQ-15IRX9:~/Downloads/Humanoid_Wato/HumanoidRL$ PYTHONPATH=$(pwd) /home/hy/IsaacLab/isaaclab.sh -p HumanoidRLPackage/rsl_rl_scripts/play.py --task=Isaac-Reach-Humanoid-Hand-v0 --num_envs=1
\ No newline at end of file
diff --git a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/reach_env_cfg.py b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/reach_env_cfg.py
index cdf5b79..00a4f2b 100644
--- a/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/reach_env_cfg.py
+++ b/HumanoidRL/HumanoidRLPackage/HumanoidRLSetup/tasks/manipulation/reach_env_cfg.py
@@ -58,16 +58,79 @@ class CommandsCfg:
 
     ee_pose = mdp.UniformPoseCommandCfg(
         asset_name="robot",
-        body_name=MISSING,
+        body_name="TIP_B_1", # end_effector name
+        resampling_time_range=(4.0, 4.0),
+        debug_vis=True,
+        ranges=mdp.UniformPoseCommandCfg.Ranges( # within task space of finger
+            pos_x=(0.07346, 0.07346),
+            pos_y=(0.13, 0.1455),
+            pos_z=(0.09, 0.1005),
+            # no rotational variation
+            roll=(0.0, 0.0),
+            pitch=(0.0, 0.0),
+            yaw=(0.0, 0.0),
+        ),
+    )
+
+    ee_pose_2 = mdp.UniformPoseCommandCfg(
+        asset_name="robot",
+        body_name="TIP_B_2",
         resampling_time_range=(4.0, 4.0),
         debug_vis=True,
         ranges=mdp.UniformPoseCommandCfg.Ranges(
-            pos_x=(0.35, 0.65),
-            pos_y=(-0.2, 0.2),
-            pos_z=(0.15, 0.5),
+            pos_x=(0.04746, 0.04746),
+            # pos_y=(0.135, 0.14),
+            pos_y=(0.15406, 0.15558),
+            # pos_z=(0.09, 0.136),
+            pos_z=(0.098, 0.10058),
             roll=(0.0, 0.0),
-            pitch=MISSING,  # depends on end-effector axis
-            yaw=(-3.14, 3.14),
+            pitch=(0.0, 0.0),
+            yaw=(0.0, 0.0),
+        ),
+    )
+
+    ee_pose_3 = mdp.UniformPoseCommandCfg(
+        asset_name="robot",
+        body_name="TIP_B_3",
+        resampling_time_range=(4.0, 4.0),
+        debug_vis=True,
+        ranges=mdp.UniformPoseCommandCfg.Ranges(
+            pos_x=(0.01996, 0.01996),
+            pos_y=(0.134, 0.145),
+            pos_z=(0.09, 0.11),
+            roll=(0.0, 0.0),
+            pitch=(0.0, 0.0),
+            yaw=(0.0, 0.0),
+        ),
+    )
+
+    ee_pose_4 = mdp.UniformPoseCommandCfg(
+        asset_name="robot",
+        body_name="TIP_B_4",
+        resampling_time_range=(4.0, 4.0),
+        debug_vis=True,
+        ranges=mdp.UniformPoseCommandCfg.Ranges(
+            pos_x=(-0.00754, -0.00754),
+            pos_y=(0.119, 0.13),
+            pos_z=(0.09, 0.1),
+            roll=(0.0, 0.0),
+            pitch=(0.0, 0.0),
+            yaw=(0.0, 0.0),
+        ),
+    )
+
+    ee_pose_5 = mdp.UniformPoseCommandCfg(
+        asset_name="robot",
+        body_name="TIP_B_5",
+        resampling_time_range=(4.0, 4.0),
+        debug_vis=True,
+        ranges=mdp.UniformPoseCommandCfg.Ranges(
+            pos_x=(0.0869, 0.0869),
+            pos_y=(0.04, 0.052),
+            pos_z=(0.17, 0.17248),
+            roll=(0.0, 0.0),
+            pitch=(0.0, 0.0),
+            yaw=(0.0, 0.0),
         ),
     )
 
@@ -91,7 +154,13 @@ class ObservationsCfg:
         # observation terms (order preserved)
         joint_pos = ObsTerm(func=mdp.joint_pos_rel, noise=Unoise(n_min=-0.01, n_max=0.01))
         joint_vel = ObsTerm(func=mdp.joint_vel_rel, noise=Unoise(n_min=-0.01, n_max=0.01))
+        
         pose_command = ObsTerm(func=mdp.generated_commands, params={"command_name": "ee_pose"})
+        pose_command_2 = ObsTerm(func=mdp.generated_commands, params={"command_name": "ee_pose_2"})
+        pose_command_3 = ObsTerm(func=mdp.generated_commands, params={"command_name": "ee_pose_3"})
+        pose_command_4 = ObsTerm(func=mdp.generated_commands, params={"command_name": "ee_pose_4"})
+        pose_command_5 = ObsTerm(func=mdp.generated_commands, params={"command_name": "ee_pose_5"})
+
         actions = ObsTerm(func=mdp.last_action)
 
         def __post_init__(self):
@@ -124,18 +193,80 @@ class RewardsCfg:
     end_effector_position_tracking = RewTerm(
         func=mdp.position_command_error,
         weight=-0.2,
-        params={"asset_cfg": SceneEntityCfg("robot", body_names=MISSING), "command_name": "ee_pose"},
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_1"), "command_name": "ee_pose"},
+    )
+    end_effector_2_position_tracking = RewTerm(
+        func=mdp.position_command_error,
+        weight=-0.2,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_2"), "command_name": "ee_pose_2"},
+    )
+    end_effector_3_position_tracking = RewTerm(
+        func=mdp.position_command_error,
+        weight=-0.2,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_3"), "command_name": "ee_pose_3"},
+    )
+    end_effector_4_position_tracking = RewTerm(
+        func=mdp.position_command_error,
+        weight=-0.2,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_4"), "command_name": "ee_pose_4"},
+    )
+    end_effector_5_position_tracking = RewTerm(
+        func=mdp.position_command_error,
+        weight=-0.2,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_5"), "command_name": "ee_pose_5"},
     )
+
     end_effector_position_tracking_fine_grained = RewTerm(
         func=mdp.position_command_error_tanh,
         weight=0.1,
-        params={"asset_cfg": SceneEntityCfg("robot", body_names=MISSING), "std": 0.1, "command_name": "ee_pose"},
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_1"), "std": 0.1, "command_name": "ee_pose"},
+    )
+    end_effector_2_position_tracking_fine_grained = RewTerm(
+        func=mdp.position_command_error_tanh,
+        weight=0.1,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_2"), "std": 0.1, "command_name": "ee_pose_2"},
+    )
+    end_effector_3_position_tracking_fine_grained = RewTerm(
+        func=mdp.position_command_error_tanh,
+        weight=0.1,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_3"), "std": 0.1, "command_name": "ee_pose_3"},
     )
-    end_effector_orientation_tracking = RewTerm(
-        func=mdp.orientation_command_error,
-        weight=-0.1,
-        params={"asset_cfg": SceneEntityCfg("robot", body_names=MISSING), "command_name": "ee_pose"},
+    end_effector_4_position_tracking_fine_grained = RewTerm(
+        func=mdp.position_command_error_tanh,
+        weight=0.1,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_4"), "std": 0.1, "command_name": "ee_pose_4"},
     )
+    end_effector_5_position_tracking_fine_grained = RewTerm(
+        func=mdp.position_command_error_tanh,
+        weight=0.1,
+        params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_5"), "std": 0.1, "command_name": "ee_pose_5"},
+    )
+
+    # end_effector_orientation_tracking = RewTerm(
+    #     func=mdp.orientation_command_error,
+    #     weight=-0.1,
+    #     params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_1"), "command_name": "ee_pose"},
+    # )
+    # end_effector_2_orientation_tracking = RewTerm(
+    #     func=mdp.orientation_command_error,
+    #     weight=-0.1,
+    #     params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_2"), "command_name": "ee_pose_2"},
+    # )
+    # end_effector_3_orientation_tracking = RewTerm(
+    #     func=mdp.orientation_command_error,
+    #     weight=-0.1,
+    #     params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_3"), "command_name": "ee_pose_3"},
+    # )
+    # end_effector_4_orientation_tracking = RewTerm(
+    #     func=mdp.orientation_command_error,
+    #     weight=-0.1,
+    #     params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_4"), "command_name": "ee_pose_4"},
+    # )
+    # end_effector_5_orientation_tracking = RewTerm(
+    #     func=mdp.orientation_command_error,
+    #     weight=-0.1,
+    #     params={"asset_cfg": SceneEntityCfg("robot", body_names="TIP_B_5"), "command_name": "ee_pose_5"},
+    # )
 
     # action penalty
     action_rate = RewTerm(func=mdp.action_rate_l2, weight=-0.0001)
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/events.out.tfevents.1748548275.hy-LOQ-15IRX9.18521.0 b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/events.out.tfevents.1748548275.hy-LOQ-15IRX9.18521.0
deleted file mode 100644
index cbb2935..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/events.out.tfevents.1748548275.hy-LOQ-15IRX9.18521.0 and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.onnx b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.onnx
deleted file mode 100644
index 873a8bb..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.onnx and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.pt
deleted file mode 100644
index d1ef491..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/exported/policy.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_0.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_0.pt
deleted file mode 100644
index 6fd2bad..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_0.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_100.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_100.pt
deleted file mode 100644
index c9ac86e..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_100.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_150.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_150.pt
deleted file mode 100644
index e018800..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_150.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_200.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_200.pt
deleted file mode 100644
index a877c2a..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_200.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_250.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_250.pt
deleted file mode 100644
index bc35c57..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_250.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_300.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_300.pt
deleted file mode 100644
index 1d5b4b7..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_300.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_350.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_350.pt
deleted file mode 100644
index 44ed6c7..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_350.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_400.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_400.pt
deleted file mode 100644
index 1abebd4..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_400.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_450.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_450.pt
deleted file mode 100644
index 6ad30c9..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_450.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_50.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_50.pt
deleted file mode 100644
index 13aaef6..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_50.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_500.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_500.pt
deleted file mode 100644
index af9be55..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_500.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_550.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_550.pt
deleted file mode 100644
index 8cbc8a4..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_550.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_600.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_600.pt
deleted file mode 100644
index 38e5edc..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_600.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_650.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_650.pt
deleted file mode 100644
index 8b1470a..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_650.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_700.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_700.pt
deleted file mode 100644
index 317984b..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_700.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_750.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_750.pt
deleted file mode 100644
index 36eb4b9..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_750.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_800.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_800.pt
deleted file mode 100644
index 4cd97e3..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_800.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_850.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_850.pt
deleted file mode 100644
index d1579c5..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_850.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_900.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_900.pt
deleted file mode 100644
index 3fde50c..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_900.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_950.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_950.pt
deleted file mode 100644
index ad23099..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_950.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_999.pt b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_999.pt
deleted file mode 100644
index b699ff1..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/model_999.pt and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.pkl b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.pkl
deleted file mode 100644
index e65cf1c..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.pkl and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.yaml b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.yaml
deleted file mode 100644
index ef4a819..0000000
--- a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/agent.yaml
+++ /dev/null
@@ -1,43 +0,0 @@
-seed: 42
-device: cuda:0
-num_steps_per_env: 24
-max_iterations: 1000
-empirical_normalization: false
-policy:
-  class_name: ActorCritic
-  init_noise_std: 1.0
-  noise_std_type: scalar
-  actor_hidden_dims:
-  - 64
-  - 64
-  critic_hidden_dims:
-  - 64
-  - 64
-  activation: elu
-algorithm:
-  class_name: PPO
-  num_learning_epochs: 8
-  num_mini_batches: 4
-  learning_rate: 0.001
-  schedule: adaptive
-  gamma: 0.99
-  lam: 0.95
-  entropy_coef: 0.01
-  desired_kl: 0.01
-  max_grad_norm: 1.0
-  value_loss_coef: 1.0
-  use_clipped_value_loss: true
-  clip_param: 0.2
-  normalize_advantage_per_mini_batch: false
-  symmetry_cfg: null
-  rnd_cfg: null
-clip_actions: null
-save_interval: 50
-experiment_name: reach_ur10
-run_name: ''
-logger: tensorboard
-neptune_project: isaaclab
-wandb_project: isaaclab
-resume: false
-load_run: .*
-load_checkpoint: model_.*.pt
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.pkl b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.pkl
deleted file mode 100644
index 7fe09d6..0000000
Binary files a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.pkl and /dev/null differ
diff --git a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.yaml b/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.yaml
deleted file mode 100644
index e13f190..0000000
--- a/HumanoidRL/logs/rsl_rl/reach_ur10/2025-05-29_15-51-10/params/env.yaml
+++ /dev/null
@@ -1,573 +0,0 @@
-viewer:
-  eye: !!python/tuple
-  - 3.5
-  - 3.5
-  - 3.5
-  lookat: !!python/tuple
-  - 0.0
-  - 0.0
-  - 0.0
-  cam_prim_path: /OmniverseKit_Persp
-  resolution: !!python/tuple
-  - 1280
-  - 720
-  origin_type: world
-  env_index: 0
-  asset_name: null
-  body_name: null
-sim:
-  physics_prim_path: /physicsScene
-  device: cuda:0
-  dt: 0.016666666666666666
-  render_interval: 2
-  gravity: !!python/tuple
-  - 0.0
-  - 0.0
-  - -9.81
-  enable_scene_query_support: false
-  use_fabric: true
-  physx:
-    solver_type: 1
-    min_position_iteration_count: 1
-    max_position_iteration_count: 255
-    min_velocity_iteration_count: 0
-    max_velocity_iteration_count: 255
-    enable_ccd: false
-    enable_stabilization: true
-    enable_enhanced_determinism: false
-    bounce_threshold_velocity: 0.5
-    friction_offset_threshold: 0.04
-    friction_correlation_distance: 0.025
-    gpu_max_rigid_contact_count: 8388608
-    gpu_max_rigid_patch_count: 163840
-    gpu_found_lost_pairs_capacity: 2097152
-    gpu_found_lost_aggregate_pairs_capacity: 33554432
-    gpu_total_aggregate_pairs_capacity: 2097152
-    gpu_collision_stack_size: 67108864
-    gpu_heap_capacity: 67108864
-    gpu_temp_buffer_capacity: 16777216
-    gpu_max_num_partitions: 8
-    gpu_max_soft_body_contacts: 1048576
-    gpu_max_particle_contacts: 1048576
-  physics_material:
-    func: isaaclab.sim.spawners.materials.physics_materials:spawn_rigid_body_material
-    static_friction: 0.5
-    dynamic_friction: 0.5
-    restitution: 0.0
-    improve_patch_friction: true
-    friction_combine_mode: average
-    restitution_combine_mode: average
-    compliant_contact_stiffness: 0.0
-    compliant_contact_damping: 0.0
-  render:
-    enable_translucency: null
-    enable_reflections: null
-    enable_global_illumination: null
-    antialiasing_mode: null
-    enable_dlssg: null
-    enable_dl_denoiser: null
-    dlss_mode: null
-    enable_direct_lighting: null
-    samples_per_pixel: null
-    enable_shadows: null
-    enable_ambient_occlusion: null
-    carb_settings: null
-    rendering_mode: null
-ui_window_class_type: isaaclab.envs.ui.manager_based_rl_env_window:ManagerBasedRLEnvWindow
-seed: 42
-decimation: 2
-scene:
-  num_envs: 4096
-  env_spacing: 2.5
-  lazy_sensor_update: true
-  replicate_physics: true
-  filter_collisions: true
-  robot:
-    class_type: isaaclab.assets.articulation.articulation:Articulation
-    prim_path: /World/envs/env_.*/Robot
-    spawn:
-      func: isaaclab.sim.spawners.from_files.from_files:spawn_from_usd
-      visible: true
-      semantic_tags: null
-      copy_from_source: true
-      mass_props: null
-      deformable_props: null
-      rigid_props:
-        rigid_body_enabled: null
-        kinematic_enabled: null
-        disable_gravity: false
-        linear_damping: null
-        angular_damping: null
-        max_linear_velocity: null
-        max_angular_velocity: null
-        max_depenetration_velocity: 5.0
-        max_contact_impulse: null
-        enable_gyroscopic_forces: null
-        retain_accelerations: null
-        solver_position_iteration_count: null
-        solver_velocity_iteration_count: null
-        sleep_threshold: null
-        stabilization_threshold: null
-      collision_props: null
-      activate_contact_sensors: false
-      scale: null
-      articulation_props: null
-      fixed_tendons_props: null
-      joint_drive_props: null
-      visual_material_path: material
-      visual_material: null
-      usd_path: http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/4.5/Isaac/IsaacLab/Robots/UniversalRobots/UR10/ur10_instanceable.usd
-      variants: null
-    init_state:
-      pos: !!python/tuple
-      - 0.0
-      - 0.0
-      - 0.0
-      rot: !!python/tuple
-      - 1.0
-      - 0.0
-      - 0.0
-      - 0.0
-      lin_vel: !!python/tuple
-      - 0.0
-      - 0.0
-      - 0.0
-      ang_vel: !!python/tuple
-      - 0.0
-      - 0.0
-      - 0.0
-      joint_pos:
-        shoulder_pan_joint: 0.0
-        shoulder_lift_joint: -1.712
-        elbow_joint: 1.712
-        wrist_1_joint: 0.0
-        wrist_2_joint: 0.0
-        wrist_3_joint: 0.0
-      joint_vel:
-        .*: 0.0
-    collision_group: 0
-    debug_vis: false
-    soft_joint_pos_limit_factor: 1.0
-    actuators:
-      arm:
-        class_type: isaaclab.actuators.actuator_pd:ImplicitActuator
-        joint_names_expr:
-        - .*
-        effort_limit: 87.0
-        velocity_limit: 100.0
-        effort_limit_sim: null
-        velocity_limit_sim: null
-        stiffness: 800.0
-        damping: 40.0
-        armature: null
-        friction: null
-  ground:
-    class_type: null
-    prim_path: /World/ground
-    spawn:
-      func: isaaclab.sim.spawners.from_files.from_files:spawn_ground_plane
-      visible: true
-      semantic_tags: null
-      copy_from_source: true
-      usd_path: http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/4.5/Isaac/Environments/Grid/default_environment.usd
-      color: !!python/tuple
-      - 0.0
-      - 0.0
-      - 0.0
-      size: !!python/tuple
-      - 100.0
-      - 100.0
-      physics_material:
-        func: isaaclab.sim.spawners.materials.physics_materials:spawn_rigid_body_material
-        static_friction: 0.5
-        dynamic_friction: 0.5
-        restitution: 0.0
-        improve_patch_friction: true
-        friction_combine_mode: average
-        restitution_combine_mode: average
-        compliant_contact_stiffness: 0.0
-        compliant_contact_damping: 0.0
-    init_state:
-      pos: !!python/tuple
-      - 0.0
-      - 0.0
-      - -1.05
-      rot: !!python/tuple
-      - 1.0
-      - 0.0
-      - 0.0
-      - 0.0
-    collision_group: 0
-    debug_vis: false
-  table:
-    class_type: null
-    prim_path: /World/envs/env_.*/Table
-    spawn:
-      func: isaaclab.sim.spawners.from_files.from_files:spawn_from_usd
-      visible: true
-      semantic_tags: null
-      copy_from_source: true
-      mass_props: null
-      deformable_props: null
-      rigid_props: null
-      collision_props: null
-      activate_contact_sensors: false
-      scale: null
-      articulation_props: null
-      fixed_tendons_props: null
-      joint_drive_props: null
-      visual_material_path: material
-      visual_material: null
-      usd_path: http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/4.5/Isaac/Props/Mounts/SeattleLabTable/table_instanceable.usd
-      variants: null
-    init_state:
-      pos: !!python/tuple
-      - 0.55
-      - 0.0
-      - 0.0
-      rot: !!python/tuple
-      - 0.70711
-      - 0.0
-      - 0.0
-      - 0.70711
-    collision_group: 0
-    debug_vis: false
-  light:
-    class_type: null
-    prim_path: /World/light
-    spawn:
-      func: isaaclab.sim.spawners.lights.lights:spawn_light
-      visible: true
-      semantic_tags: null
-      copy_from_source: true
-      prim_type: DomeLight
-      color: !!python/tuple
-      - 0.75
-      - 0.75
-      - 0.75
-      enable_color_temperature: false
-      color_temperature: 6500.0
-      normalize: false
-      exposure: 0.0
-      intensity: 2500.0
-      texture_file: null
-      texture_format: automatic
-      visible_in_primary_ray: true
-    init_state:
-      pos: !!python/tuple
-      - 0.0
-      - 0.0
-      - 0.0
-      rot: !!python/tuple
-      - 1.0
-      - 0.0
-      - 0.0
-      - 0.0
-    collision_group: 0
-    debug_vis: false
-recorders:
-  dataset_file_handler_class_type: isaaclab.utils.datasets.hdf5_dataset_file_handler:HDF5DatasetFileHandler
-  dataset_export_dir_path: /tmp/isaaclab/logs
-  dataset_filename: dataset
-  dataset_export_mode:
-    _value_: 1
-    _name_: EXPORT_ALL
-  export_in_record_pre_reset: true
-observations:
-  policy:
-    concatenate_terms: true
-    enable_corruption: true
-    history_length: null
-    flatten_history_dim: true
-    joint_pos:
-      func: isaaclab.envs.mdp.observations:joint_pos_rel
-      params: {}
-      modifiers: null
-      noise:
-        func: isaaclab.utils.noise.noise_model:uniform_noise
-        operation: add
-        n_min: -0.01
-        n_max: 0.01
-      clip: null
-      scale: null
-      history_length: 0
-      flatten_history_dim: true
-    joint_vel:
-      func: isaaclab.envs.mdp.observations:joint_vel_rel
-      params: {}
-      modifiers: null
-      noise:
-        func: isaaclab.utils.noise.noise_model:uniform_noise
-        operation: add
-        n_min: -0.01
-        n_max: 0.01
-      clip: null
-      scale: null
-      history_length: 0
-      flatten_history_dim: true
-    pose_command:
-      func: isaaclab.envs.mdp.observations:generated_commands
-      params:
-        command_name: ee_pose
-      modifiers: null
-      noise: null
-      clip: null
-      scale: null
-      history_length: 0
-      flatten_history_dim: true
-    actions:
-      func: isaaclab.envs.mdp.observations:last_action
-      params: {}
-      modifiers: null
-      noise: null
-      clip: null
-      scale: null
-      history_length: 0
-      flatten_history_dim: true
-actions:
-  arm_action:
-    class_type: isaaclab.envs.mdp.actions.joint_actions:JointPositionAction
-    asset_name: robot
-    debug_vis: false
-    clip: null
-    joint_names:
-    - .*
-    scale: 0.5
-    offset: 0.0
-    preserve_order: false
-    use_default_offset: true
-  gripper_action: null
-events:
-  reset_robot_joints:
-    func: isaaclab.envs.mdp.events:reset_joints_by_scale
-    params:
-      position_range: !!python/tuple
-      - 0.75
-      - 1.25
-      velocity_range: !!python/tuple
-      - 0.0
-      - 0.0
-    mode: reset
-    interval_range_s: null
-    is_global_time: false
-    min_step_count_between_reset: 0
-rerender_on_reset: false
-wait_for_textures: true
-xr: null
-is_finite_horizon: false
-episode_length_s: 12.0
-rewards:
-  end_effector_position_tracking:
-    func: HumanoidRLPackage.HumanoidRLSetup.tasks.manipulation.mdp.rewards:position_command_error
-    params:
-      asset_cfg:
-        name: robot
-        joint_names: null
-        joint_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        fixed_tendon_names: null
-        fixed_tendon_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        body_names:
-        - ee_link
-        body_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        object_collection_names: null
-        object_collection_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        preserve_order: false
-      command_name: ee_pose
-    weight: -0.2
-  end_effector_position_tracking_fine_grained:
-    func: import HumanoidRLPackage.HumanoidRLSetup.tasks.manipulation.mdp.rewards:position_command_error_tanh
-    params:
-      asset_cfg:
-        name: robot
-        joint_names: null
-        joint_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        fixed_tendon_names: null
-        fixed_tendon_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        body_names:
-        - ee_link
-        body_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        object_collection_names: null
-        object_collection_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        preserve_order: false
-      std: 0.1
-      command_name: ee_pose
-    weight: 0.1
-  end_effector_orientation_tracking:
-    func: import HumanoidRLPackage.HumanoidRLSetup.tasks.manipulation.mdp.rewards:orientation_command_error
-    params:
-      asset_cfg:
-        name: robot
-        joint_names: null
-        joint_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        fixed_tendon_names: null
-        fixed_tendon_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        body_names:
-        - ee_link
-        body_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        object_collection_names: null
-        object_collection_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        preserve_order: false
-      command_name: ee_pose
-    weight: -0.1
-  action_rate:
-    func: isaaclab.envs.mdp.rewards:action_rate_l2
-    params: {}
-    weight: -0.0001
-  joint_vel:
-    func: isaaclab.envs.mdp.rewards:joint_vel_l2
-    params:
-      asset_cfg:
-        name: robot
-        joint_names: null
-        joint_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        fixed_tendon_names: null
-        fixed_tendon_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        body_names: null
-        body_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        object_collection_names: null
-        object_collection_ids: !!python/object/apply:builtins.slice
-        - null
-        - null
-        - null
-        preserve_order: false
-    weight: -0.0001
-terminations:
-  time_out:
-    func: isaaclab.envs.mdp.terminations:time_out
-    params: {}
-    time_out: true
-curriculum:
-  action_rate:
-    func: isaaclab.envs.mdp.curriculums:modify_reward_weight
-    params:
-      term_name: action_rate
-      weight: -0.005
-      num_steps: 4500
-  joint_vel:
-    func: isaaclab.envs.mdp.curriculums:modify_reward_weight
-    params:
-      term_name: joint_vel
-      weight: -0.001
-      num_steps: 4500
-commands:
-  ee_pose:
-    class_type: isaaclab.envs.mdp.commands.pose_command:UniformPoseCommand
-    resampling_time_range: !!python/tuple
-    - 4.0
-    - 4.0
-    debug_vis: true
-    asset_name: robot
-    body_name: ee_link
-    make_quat_unique: false
-    ranges:
-      pos_x: !!python/tuple
-      - 0.35
-      - 0.65
-      pos_y: !!python/tuple
-      - -0.2
-      - 0.2
-      pos_z: !!python/tuple
-      - 0.15
-      - 0.5
-      roll: !!python/tuple
-      - 0.0
-      - 0.0
-      pitch: !!python/tuple
-      - 1.5707963267948966
-      - 1.5707963267948966
-      yaw: !!python/tuple
-      - -3.14
-      - 3.14
-    goal_pose_visualizer_cfg:
-      prim_path: /Visuals/Command/goal_pose
-      markers:
-        frame:
-          func: isaaclab.sim.spawners.from_files.from_files:spawn_from_usd
-          visible: true
-          semantic_tags: null
-          copy_from_source: true
-          mass_props: null
-          deformable_props: null
-          rigid_props: null
-          collision_props: null
-          activate_contact_sensors: false
-          scale: !!python/tuple
-          - 0.1
-          - 0.1
-          - 0.1
-          articulation_props: null
-          fixed_tendons_props: null
-          joint_drive_props: null
-          visual_material_path: material
-          visual_material: null
-          usd_path: http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/4.5/Isaac/Props/UIElements/frame_prim.usd
-          variants: null
-    current_pose_visualizer_cfg:
-      prim_path: /Visuals/Command/body_pose
-      markers:
-        frame:
-          func: isaaclab.sim.spawners.from_files.from_files:spawn_from_usd
-          visible: true
-          semantic_tags: null
-          copy_from_source: true
-          mass_props: null
-          deformable_props: null
-          rigid_props: null
-          collision_props: null
-          activate_contact_sensors: false
-          scale: !!python/tuple
-          - 0.1
-          - 0.1
-          - 0.1
-          articulation_props: null
-          fixed_tendons_props: null
-          joint_drive_props: null
-          visual_material_path: material
-          visual_material: null
-          usd_path: http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/4.5/Isaac/Props/UIElements/frame_prim.usd
-          variants: null
diff --git a/ModelAssets/.thumbs/256x256/hand.usd.png b/ModelAssets/.thumbs/256x256/hand.usd.png
deleted file mode 100644
index 65b14aa..0000000
Binary files a/ModelAssets/.thumbs/256x256/hand.usd.png and /dev/null differ
diff --git a/ModelAssets/configuration/hand_base.usd b/ModelAssets/configuration/hand_base.usd
index c9e4c91..580fe86 100644
Binary files a/ModelAssets/configuration/hand_base.usd and b/ModelAssets/configuration/hand_base.usd differ
diff --git a/ModelAssets/configuration/hand_physics.usd b/ModelAssets/configuration/hand_physics.usd
index 40153dc..96c77c4 100644
Binary files a/ModelAssets/configuration/hand_physics.usd and b/ModelAssets/configuration/hand_physics.usd differ
diff --git a/ModelAssets/configuration/hand_sensor.usd b/ModelAssets/configuration/hand_sensor.usd
index e401420..35a9fe2 100644
Binary files a/ModelAssets/configuration/hand_sensor.usd and b/ModelAssets/configuration/hand_sensor.usd differ
diff --git a/ModelAssets/hand.usd b/ModelAssets/hand.usd
index 2344211..2bc943c 100644
Binary files a/ModelAssets/hand.usd and b/ModelAssets/hand.usd differ
diff --git a/hand_test.py b/hand_test.py
index e02d8c7..d4b8eeb 100644
--- a/hand_test.py
+++ b/hand_test.py
@@ -56,21 +56,34 @@ class HandSceneCfg(InteractiveSceneCfg):
 
 def run_simulator(sim: sim_utils.SimulationContext, scene: InteractiveScene):
     
-    # robot = scene["robot"]
+    robot = scene["robot"]
 
     # robot_entity_cfg = SceneEntityCfg("robot", joint_names=[".*"], body_names=["ee_link"])
     # robot_entity_cfg = SceneEntityCfg("robot", joint_names=["panda_joint.*"], body_names=["panda_hand"])
+    robot_entity_cfg = SceneEntityCfg("robot", joint_names=[".*"], body_names=[".*"])
 
-    # robot_entity_cfg.resolve(scene)
+    robot_entity_cfg.resolve(scene)
 
     sim_dt = sim.get_physics_dt()
 
+    joint_position = robot.data.default_joint_pos.clone()
+    joint_vel = robot.data.default_joint_vel.clone()
+    robot.write_joint_state_to_sim(joint_position, joint_vel)
+
     while simulation_app.is_running():
 
         # joint_position = robot.data.default_joint_pos.clone()
         # print(joint_position)
         # joint_vel = robot.data.default_joint_vel.clone()
         # robot.write_joint_state_to_sim(joint_position, joint_vel)
+        
+        joint_position_list = [[1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]]
+        joint_position = torch.tensor(joint_position_list[0], device=sim.device)
+
+        robot.reset()
+        joint_pos_des = joint_position.unsqueeze(0)[:, robot_entity_cfg.joint_ids].clone()
+        robot.set_joint_position_target(joint_pos_des, joint_ids=robot_entity_cfg.joint_ids)
+        scene.write_data_to_sim()
 
         sim.step()
 
diff --git a/test_old.py b/test_old.py
deleted file mode 100644
index 79161cd..0000000
--- a/test_old.py
+++ /dev/null
@@ -1,187 +0,0 @@
-import argparse
-from isaaclab.app import AppLauncher
-
-parser = argparse.ArgumentParser(description="Robot Arm with Task Space IK Control")
-parser.add_argument("--robot", type=str, default="franka_panda", help="Name of the robot.")
-AppLauncher.add_app_launcher_args(parser)
-args_cli = parser.parse_args()
-
-app_launcher = AppLauncher(args_cli)
-simulation_app = app_launcher.app
-
-import torch
-
-import isaaclab.sim as sim_utils
-from isaaclab.assets import AssetBaseCfg
-from isaaclab.controllers import DifferentialIKController, DifferentialIKControllerCfg
-from isaaclab.managers import SceneEntityCfg
-from isaaclab.markers import VisualizationMarkers
-from isaaclab.markers.config import FRAME_MARKER_CFG
-from isaaclab.scene import InteractiveScene, InteractiveSceneCfg
-from isaaclab.utils import configclass
-from isaaclab.utils.assets import ISAAC_NUCLEUS_DIR
-from isaaclab.utils.math import subtract_frame_transforms
-
-##
-# Pre-defined configs
-##
-from isaaclab_assets import UR10_CFG, FRANKA_PANDA_HIGH_PD_CFG
-
-
-@configclass
-class TableTopSceneCfg(InteractiveSceneCfg):
-    """Configuration for a cart-pole scene."""
-
-    # ground plane
-    ground = AssetBaseCfg(
-        prim_path="/World/defaultGroundPlane",
-        spawn=sim_utils.GroundPlaneCfg(),
-        init_state=AssetBaseCfg.InitialStateCfg(pos=(0.0, 0.0, -1.05)),
-    )
-
-    # lights
-    dome_light = AssetBaseCfg(
-        prim_path="/World/Light",
-        spawn=sim_utils.DomeLightCfg(intensity=3000.0, color=(0.75, 0.75, 0.75))
-    )
-
-    # mount
-    table = AssetBaseCfg(
-        prim_path="{ENV_REGEX_NS}/Table",
-        spawn=sim_utils.UsdFileCfg(
-            usd_path=f"{ISAAC_NUCLEUS_DIR}/Props/Mounts/Stand/stand_instanceable.usd", scale=(2.0, 2.0, 2.0)
-        ),
-    )
-
-    # Cube
-    cube = AssetBaseCfg(
-        prim_path="/World/cube",
-        spawn=sim_utils.CuboidCfg(
-            size=[0.1, 0.1, 0.1]
-        ),
-        init_state=AssetBaseCfg.InitialStateCfg(pos=(0.5, 0.0, 0.5)),
-    )
-
-    # articulation
-    if args_cli.robot == "franka_panda":
-        robot = FRANKA_PANDA_HIGH_PD_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
-    elif args_cli.robot == "ur10":
-        robot = UR10_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
-    else:
-        raise ValueError(f"Robot {args_cli.robot} is not supported. Valid: franka_panda, ur10")
-    # robot = UR10_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
-
-
-def run_simulator(sim: sim_utils.SimulationContext, scene: InteractiveScene):
-    
-    robot = scene["robot"]
-
-    # Create controller
-    diff_ik_cfg = DifferentialIKControllerCfg(command_type="pose", use_relative_mode=False, ik_method="dls")
-    diff_ik_controller = DifferentialIKController(diff_ik_cfg, num_envs=scene.num_envs, device=sim.device)
-
-    # Markers
-    frame_marker_cfg = FRAME_MARKER_CFG.copy()
-    frame_marker_cfg.markers["frame"].scale = (0.1, 0.1, 0.1)
-    ee_marker = VisualizationMarkers(frame_marker_cfg.replace(prim_path="/Visuals/ee_current"))
-    goal_marker = VisualizationMarkers(frame_marker_cfg.replace(prim_path="/Visuals/ee_goal"))
-
-    # Specify robot-specific parameters
-    if args_cli.robot == "franka_panda":
-        robot_entity_cfg = SceneEntityCfg("robot", joint_names=["panda_joint.*"], body_names=["panda_hand"])
-    elif args_cli.robot == "ur10":
-        robot_entity_cfg = SceneEntityCfg("robot", joint_names=[".*"], body_names=["ee_link"])
-    else:
-        raise ValueError(f"Robot {args_cli.robot} is not supported. Valid: franka_panda, ur10")
-    # robot_entity_cfg = SceneEntityCfg("robot", joint_names=[".*"], body_names=["ee_link"])
-
-    # Resolving the scene entities
-    robot_entity_cfg.resolve(scene)
-
-    # Obtain the frame index of the end-effector ; For a fixed base robot, the frame index is one less than the body index. This is because ; the root body is not included in the returned Jacobians.
-    if robot.is_fixed_base:
-        ee_jacobi_idx = robot_entity_cfg.body_ids[0] - 1
-    else:
-        ee_jacobi_idx = robot_entity_cfg.body_ids[0]
-
-    # Define simulation stepping
-    sim_dt = sim.get_physics_dt()
-
-    # May have to set initial position first
-    if args_cli.robot == "franka_panda":
-        joint_position = robot.data.default_joint_pos.clone()
-        joint_vel = robot.data.default_joint_vel.clone()
-        robot.write_joint_state_to_sim(joint_position, joint_vel)
-    else:
-        joint_position_index = 0
-        joint_position_list = [
-            [0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
-        ]
-        joint_position = torch.tensor(joint_position_list[joint_position_index], device=sim.device)
-        joint_vel = robot.data.default_joint_vel.clone()
-        joint_pos_des = joint_position.unsqueeze(0)[:, robot_entity_cfg.joint_ids].clone()
-        robot.write_joint_state_to_sim(joint_pos_des, joint_vel)
-
-    while simulation_app.is_running():
-        # Get cube/target_point coordinates
-        position, quaternion = scene["cube"].get_world_poses()
-        ik_commands = torch.cat([position, quaternion], dim=1) # Quaternion is in (w, x, y, z)
-
-        diff_ik_controller.set_command(ik_commands)
-
-        # obtain quantities from simulation
-        jacobian = robot.root_physx_view.get_jacobians()[:, ee_jacobi_idx, :, robot_entity_cfg.joint_ids]
-        ee_pose_w = robot.data.body_state_w[:, robot_entity_cfg.body_ids[0], 0:7]
-        root_pose_w = robot.data.root_state_w[:, 0:7]
-        joint_pos = robot.data.joint_pos[:, robot_entity_cfg.joint_ids]
-
-        # compute frame in root frame
-        ee_pos_b, ee_quat_b = subtract_frame_transforms(
-            root_pose_w[:, 0:3], root_pose_w[:, 3:7], ee_pose_w[:, 0:3], ee_pose_w[:, 3:7]
-        )
-
-        # compute the joint commands
-        joint_pos_des = diff_ik_controller.compute(ee_pos_b, ee_quat_b, jacobian, joint_pos)
-
-        # apply actions (Smooth movement)
-        robot.set_joint_position_target(joint_pos_des, joint_ids=robot_entity_cfg.joint_ids)
-        scene.write_data_to_sim()
-
-        # perform step
-        sim.step()
-
-        # update buffers
-        scene.update(sim_dt)
-
-        # obtain quantities from simulation
-        ee_pose_w = robot.data.body_state_w[:, robot_entity_cfg.body_ids[0], 0:7]
-
-        # update marker positions
-        ee_marker.visualize(ee_pose_w[:, 0:3], ee_pose_w[:, 3:7])
-        goal_marker.visualize(ik_commands[:, 0:3] + scene.env_origins, ik_commands[:, 3:7])
-
-
-def main():
-    # Load kit helper
-    sim_cfg = sim_utils.SimulationCfg(dt=0.01, device=args_cli.device)
-    sim = sim_utils.SimulationContext(sim_cfg)
-
-    # Set main camera
-    sim.set_camera_view([2.5, 2.5, 2.5], [0.0, 0.0, 0.0])
-
-    # Design scene
-    scene_cfg = TableTopSceneCfg(num_envs=1, env_spacing=2.0)
-    scene = InteractiveScene(scene_cfg)
-
-    # Play the simulator
-    sim.reset()
-
-    print("[INFO]: Setup complete...")
-
-    # Run the simulator
-    run_simulator(sim, scene)
-
-
-if __name__ == "__main__":
-    main()
-    simulation_app.close()
\ No newline at end of file